test/extensions/filters/http/lua/wrappers_test.cc

```diff
@@ -311,6 +311,81 @@ TEST_F(LuaStreamInfoWrapperTest, SetGetAndIterateDynamicMetadata) {
   wrapper.reset();
 }
 
+// Set, get complex key/values in stream info dynamic metadata.
```

Comment:
Mind to extend one in lua_filter_test and lua_integration_test? Most of the time takin a look at these two tests is the best way to learn on the usability of it.